schema/schema.hh

```diff
@@ -161,7 +161,11 @@ private:
     type _t;
     double _v;
 public:
-    speculative_retry(type t, double v) : _t(t), _v(v) {}
+    speculative_retry(type t, double v) : _t(t), _v(v) {
+        if (t == type::PERCENTILE && (v < 0 || v > 1.0)) {
+            throw std::invalid_argument(format("invalid percentile value: {:f}. Value must be between 0 and 1 inclusive\n", v));
+        }
```

Comment:
I think this is not necessary, you already validated the value in `from_string()`.

Reply:
But no one guarantees that this public constructor will be called only with valid parameters, which can cause unpleasant side effects.

Reply:
Ok, this is a bit too defensive for my taste, but I won't insist.

Reply:
Hi. I listened to your opinion and did the validation only in the factory method.